rpc/executables.go

```diff
@@ -0,0 +1,155 @@
+package rpc
+
+import (
+	"encoding/json"
+
+	hintRunnerZero "github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/zero"
+	"github.com/NethermindEth/cairo-vm-go/pkg/parsers/zero"
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/NethermindEth/juno/utils"
+)
+
+type CasmEntryPoint struct {
+	Offset   *felt.Felt `json:"offset"`
+	Selector *felt.Felt `json:"selector"`
+	Builtins []string   `json:"builtins"`
+}
+
+type EntryPointsByType struct {
+	Constructor []CasmEntryPoint `json:"CONSTRUCTOR"`
+	External    []CasmEntryPoint `json:"EXTERNAL"`
+	L1Handler   []CasmEntryPoint `json:"L1_HANDLER"`
+}
+
+type CasmCompiledContractClass struct {
+	EntryPointsByType EntryPointsByType `json:"entry_points_by_type"`
+	// can't use felt.Felt here because prime is larger than felt
+	Prime                  string          `json:"prime"`
+	CompilerVersion        string          `json:"compiler_version"`
+	Bytecode               []*felt.Felt    `json:"bytecode"`
+	Hints                  json.RawMessage `json:"hints"`
+	BytecodeSegmentLengths []int           `json:"bytecode_segment_lengths,omitempty"`
+}
+
+func (h *Handler) CompiledCasm(classHash *felt.Felt) (*CasmCompiledContractClass, *jsonrpc.Error) {
+	state, stateCloser, err := h.bcReader.HeadState()
+	if err != nil {
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+	}
+	defer h.callAndLogErr(stateCloser, "failed to close state reader")
+
+	declaredClass, err := state.Class(classHash)
```

Comment:
Shouldn't we return [`CLASS_HASH_NOT_FOUND`](https://github.com/starkware-libs/starknet-specs/blob/19ab9d4df4ae3acc1a52cde5a43a7cace08bcc4b/api/starknet_executables.json#L37C50-L37C70) in case class not found?

Reply:
good point, thanks